### PR TITLE
RUN protocol and Pegasus via governance

### LINKED ISSUE #5029: RUN bootstrap packaging does not support post-genesis deployment
## Describe the bug

The bootstrap code for runStake is in econ-behaviors.js now, but that code post-dates agoricdev-8, so to deploy runStake to the current devnet (#4348) I had to re-package the relevant code (including stuff like `AmountMath.make`) as an evaluatable script. The result was 100 or so lines of code with no automated tests.

Deploying contracts that weren't available at genesis is more likely the rule than the exception, so we should figure out a maintainable way to do it.

## To Reproduce
n/a?

## Expected behavior

governance proposals to start run protocol contracts should consist of no more than a few lines of JS that hasn't been through automated tests.

## Platform Environment

agoricdev8
more details are in #4348

## Additional context

@michaelfig and I started brainstorming ideas involving the endo bundler, endoCAS, integration tests similar to agoric-cli, etc.

### LINKED ISSUE #5128: economy missing from sim-chain, loadgen
## Describe the bug

The AMM etc. are not present in the sim chain nor the loadgen config as of #5123.

## To Reproduce

_I'm not sure about the first few steps; I got the symptoms below from @michaelfig in discussion with @mhofman ..._

1. run local-chain
2. start loadgen
3. See timeout:

```
stage-0: loadgen: Load gen app stopped successfully
stage-0: runner: Stopping client
stage-0: client: 2022-04-16T18:28:06.052Z shutdown: Shutting down cleanly...
stage-0: client: 2022-04-16T18:28:06.052Z shutdown: Process terminated!
stage-0: client: Client exited successfully
stage-0: runner: Stopping chain
stage-0: chain: Chain exited successfully
stage-0: runner: Live blocks stats: {
stage-0: runner:   blockCount: 131,
stage-0: runner:   liveMode: true,
stage-0: runner:   startBlockHeight: 30,
stage-0: runner:   endBlockHeight: 160,
stage-0: runner:   avgLag: 6.23992,
stage-0: runner:   avgBlockDuration: 5.42943,
stage-0: runner:   avgChainBlockDuration: 5.442748,
stage-0: runner:   avgIdleTime: 5.197447,
stage-0: runner:   avgCosmosTime: 0.002758,
stage-0: runner:   avgSwingsetTime: 0.229224,
stage-0: runner:   avgProcessingTime: 0.231982,
stage-0: runner:   avgDeliveries: 6.824427,
stage-0: runner:   avgComputrons: 861199.748092,
stage-0: runner:   avgSwingsetPercentage: 3.76,
stage-0: runner:   avgProcessingPercentage: 3.81,
stage-0: runner:   p95Lag: 7.072706,
stage-0: runner:   p95BlockDuration: 6.149642,
stage-0: runner:   p95ChainBlockDuration: 6,
stage-0: runner:   p95IdleTime: 5.412414,
stage-0: runner:   p95CosmosTime: 0.008152,
stage-0: runner:   p95SwingsetTime: 1.15307,
stage-0: runner:   p95ProcessingTime: 1.154879,
stage-0: runner:   p95Deliveries: 35,
stage-0: runner:   p95Computrons: 1291399,
stage-0: runner:   p95SwingsetPercentage: 18.94,
stage-0: runner:   p95ProcessingPercentage: 19.29
stage-0: runner: }
stage-0: runner: Cycles stats: { cycleCount: 0 }
init: Saving chain storage
init: Saving client storage
summary: Waiting for storage backups to finish
summary: Live blocks stats: { blockCount: 0 }
summary: Cycles stats: { cycleCount: 0 }
(Error#1)
Error#1: Timeout

  at file:///usr/src/testnet-load-generator/runner/lib/helpers/async.js:100:22
  at async file:///usr/src/testnet-load-generator/runner/lib/main.js:686:11
  at async spawnLoadgen (file:///usr/src/testnet-load-generator/runner/lib/main.js:684:7)
  at async file:///usr/src/testnet-load-generator/runner/lib/helpers/async.js:129:5
  at async file:///usr/src/testnet-load-generator/runner/lib/helpers/async.js:130:7
  at async file:///usr/src/testnet-load-generator/runner/lib/main.js:629:11
  at async spawnClient (file:///usr/src/testnet-load-generator/runner/lib/main.js:611:7)
  at async file:///usr/src/testnet-load-generator/runner/lib/helpers/async.js:129:5
  at async file:///usr/src/testnet-load-generator/runner/lib/helpers/async.js:130:7
  at async file:///usr/src/testnet-load-generator/runner/lib/main.js:560:11
  at async spawnChain (file:///usr/src/testnet-load-generator/runner/lib/main.js:536:7)
  at async file:///usr/src/testnet-load-generator/runner/lib/helpers/async.js:129:5
  at async file:///usr/src/testnet-load-generator/runner/lib/helpers/async.js:130:7
  at async rootTask (file:///usr/src/testnet-load-generator/runner/lib/main.js:780:11)
  at async file:///usr/src/testnet-load-generator/runner/lib/helpers/async.js:129:5
  at async file:///usr/src/testnet-load-generator/runner/lib/main.js:800:9
  at async runStage (file:///usr/src/testnet-load-generator/runner/lib/main.js:774:5)
  at async logPerfEvent.stats (file:///usr/src/testnet-load-generator/runner/lib/main.js:1015:9)
  at async main (file:///usr/src/testnet-load-generator/runner/lib/main.js:825:3)

Error: Process completed with exit code 2.
```

## Expected behavior

sim chain provides AMM, vaultFactory instance, wired up for trading.
loadgen works.

## Platform Environment
_IOU_
 - what OS are you using? what version of Node.js?
 - is there anything special/unusual about your platform?
 - what version of the Agoric-SDK are you using? (run `git describe --tags --always`)

## Additional context

The upcoming devnet #5049 is the reason for landing #5123 despite the regression.

### PR DESCRIPTION
**Pull request by michaelfig**
closes: #5029
closes: #5128

refs: #5049

## Description

Restore the economy and Pegasus to the sim-chain, loadgen etc.

Core proposals are proposals for use with swingset-core-eval. In production, they are triggered by BLD holder governance decisions, but for sim-chain and such, they can be declared statically in the chain configuration, in which case they are run at bootstrap.
 
future work idea:
 - [ ] separate cosmos proposal items rather than merged permits

### Documentation Considerations

 - [ ] `agoric deploy` now resolves package specifiers as well as paths.

### Testing Considerations

nice to have:
 - [ ] test for installInPieces fix - integration test of installing RUN protocol contracts on a local-chain
 - [x] `producer.reset` unit test

### CODE REVIEW

File: packages/pegasus/scripts/init-core.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import { makeHelpers } from '@agoric/deploy-script-support';
+
+export const defaultProposalBuilder = async ({ publishRef, install }) =>
```
Comment (dckc) [inline packages/pegasus/scripts/init-core.js:3]: here's hoping we find time to write docs for this. not essential today.

File: packages/deploy-script-support/src/coreProposalBehavior.js [hunk 2]
```diff
@@ -84,3 +87,32 @@ export const makeCoreProposalBehavior = ({
   // Make the behavior the completion value.
   return behavior;
 };
+
+export const makeEnactCoreProposals =
```
Comment (dckc) [inline packages/deploy-script-support/src/coreProposalBehavior.js:91]: might be nice to document that this is not a production code path

File: packages/run-protocol/scripts/init-core.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
+/* global process */
+import { makeHelpers } from '@agoric/deploy-script-support';
+
+export const defaultProposalBuilder = async ({ publishRef, install }) => {
```
Comment (dckc) [inline packages/run-protocol/scripts/init-core.js:4]: Note: we can continue to tweak this until we actually deploy the proposal (at the end of the week).

File: packages/run-protocol/src/core-proposal.js [hunk 4]
```diff
@@ -2,27 +2,14 @@ export * from './econ-behaviors.js';
 export * from './sim-behaviors.js';
 
 const SHARED_POST_BOOT_MANIFEST = harden({
```
Comment (dckc) [inline packages/run-protocol/src/core-proposal.js:4]: This isn't as easy to maintain, now that it's out of scope of `test-boot.js` in `packages/vats`.

File: packages/vats/src/core/boot.js [hunk 5]
```diff
@@ -107,6 +107,28 @@ const buildRootObject = (vatPowers, vatParameters) => {
     });
 
     await runBehaviors(bootManifest);
+
+    const { coreProposalCode } = vatParameters;
+    if (!coreProposalCode) {
```
Comment (dckc) [inline packages/vats/src/core/boot.js:112]: wish: completely separate boot.js for production and non-production, with no conditionals like this

**Review by dckc — COMMENTED** [review]
Looks good overall.

In discussion, @michaelfig and I found it worth adding docs for "core proposal" and adding a unit test. When those are in, I plan to approve.

**Review by dckc — APPROVED** [review]
glad I asked for the test :)

File: packages/run-protocol/src/econ-behaviors.js [hunk 6]
```diff
@@ -330,25 +329,21 @@ export const startVaultFactory = async (
...
  * Grant access to the VaultFactory creatorFacet
  * to up to one user based on address.
  *
- * @param { EconomyBootstrapPowers & {
- *   vatParameters: { argv: { vaultFactoryControllerAddress?: string } }
- * }} powers
+ * @param { EconomyBootstrapPowers } powers
+ * @param {object} [root0]
+ * @param {object} [root0.options]
+ * @param {string} [root0.options.vaultFactoryControllerAddress]
```
Comment (turadg) [inline packages/run-protocol/src/econ-behaviors.js:341]: `root0` looks like noise. consider `args` or `config`

File: packages/run-protocol/src/econ-behaviors.js [hunk 7]
```diff
@@ -574,14 +568,14 @@ export const startRunStake = async (
...
-    recordingPeriod: SECONDS_PER_DAY,
-  },
+  {
+    debtLimit = 1_000_000_000_000n,
+    mintingRatio = [1n, 4n],
+    interestRateBP = 250n,
+    loanFeeBP = 200n,
+    chargingPeriod = SECONDS_PER_HOUR,
+    recordingPeriod = SECONDS_PER_DAY,
+  } = {},
```
Comment (turadg) [inline packages/run-protocol/src/econ-behaviors.js:578]: I take it that the intent is if a partial object is provided that its values will come through and the parts that aren't defined will get these defaults. I haven't see that before so I'll trust that you have confirmed that works.

File: packages/run-protocol/src/core-proposal.js [hunk 8]
```diff
@@ -195,20 +179,25 @@ const roleToGovernanceActions = harden({
...
-  { ROLE = 'chain', installKeys },
+  { ROLE = 'chain', installKeys, vaultFactoryControllerAddress },
 ) => {
   return {
     manifest: roleToGovernanceActions[ROLE],
     installations: {
       runStake: restoreRef(installKeys.runStake),
       amm: restoreRef(installKeys.amm),
-      vaultFactory: restoreRef(installKeys.vaultFactory),
+      VaultFactory: restoreRef(installKeys.vaultFactory),
```
Comment (turadg) [inline packages/run-protocol/src/core-proposal.js:191]: capitalization change necessary?

**Review by turadg — APPROVED** [review]
Looked at `run-protocol`. Some concerns but nothing blocking.

File: packages/run-protocol/src/econ-behaviors.js (hunk 7 above)
Comment (mhofman) [reply packages/run-protocol/src/econ-behaviors.js:578]: Yes that's how default assignment in any destructure works. I've used it plenty of times, can confirm it works :)

File: packages/run-protocol/src/core-proposal.js (hunk 8 above)
Comment (dckc) [reply packages/run-protocol/src/core-proposal.js:191]: this change is necessary for consistency with other places.

Perhaps they should all be changed to `vaultFactory`, but that involves the dApps too, so... not today.

File: packages/run-protocol/src/econ-behaviors.js (hunk 6 above)
Comment (dckc) [reply packages/run-protocol/src/econ-behaviors.js:341]: yes, some robot picked that. good catch. unfortunately we rushed a bit an missed this comment. Here's hoping to clean it up in due course...

**Pull request merged**